src/components/cuda/linux-cuda.c

```diff
@@ -1,5651 +1,442 @@
+    // Initialize global_event_names array
+    global_event_names = initialize_dynamic_event_list();
+    if (global_event_names == NULL) {
+        papi_errno = PAPI_ENOMEM;
+        goto fn_exit;
     }
- 
-    // We need the rawMetricRequests, which we collected during event enumeration.
-    // We also need the nvidia names of the metrics: cuda11_AllEvents[]->nv_name.
-    // We need to assemble by device, and one array for all events on that device.
-   
-    int *deviceMetricCount =    calloc(gctxt->deviceCount, sizeof(int));
-    int *deviceRawMetricCount = calloc(gctxt->deviceCount, sizeof(int));
-
-    // Note, in cuda11, we do not use gctxt->availEventIsBeingMeasuredInEventset[idx].
-    // Always reset to zero. We rebuild the eventset from scratch on every call.
-    gctrl->activeEventCount=0;
 
-    // First up: ensure every event is populated.
-    cuda_device_desc_t *mydevice;
-    for (ii = 0; ii < nativeCount; ii++) {
-        int idx = nativeInfo[ii].ni_event;
-        // skip if already initialized.
-        if (cuda11_AllEvents[idx]->detailsDone == 1) continue;
-        dev = cuda11_AllEvents[idx]->deviceNum;
-        // get or create the appropriate Metrics Context.
-        NVPW_CUDA_MetricsContext_Create_Params *pMCCP = cuda11_getMetricsContextPtr(dev);
-        mydevice = &gctxt->deviceArray[dev];
-        int err  = cuda11_getMetricDetails(cuda11_AllEvents[idx], mydevice->cuda11_chipName, pMCCP);
-        if (err != PAPI_OK) {
-            if (0) fprintf(stderr, "%s:%s:%i cuda11_getMetricDetails() failed, index=%d err=%d '%s'.\n",
-                     __FILE__, __func__, __LINE__, idx, err, PAPI_strerror(err));
-            return(err);
-        }
+    papi_errno = cuptid_init(&disabled_reason);
+    if (papi_errno != PAPI_OK)
+    {
+        sprintf(_cuda_vector.cmp_info.disabled_reason, disabled_reason);
+        _cuda_vector.cmp_info.disabled = papi_errno;
+        goto fn_exit;
     }
 
-    for (ii = 0; ii < nativeCount; ii++) {                                  // For each event provided by caller,
-        nativeInfo[ii].ni_position = gctrl->activeEventCount++; 
-        int idx = nativeInfo[ii].ni_event;                                  // Get the index of the event (in the global context).
-        gctrl->activeEventIndex[ii] = idx;                                  // Remember global index for this value.
-        // Here we init the values for this event.
-        switch (cuda11_AllEvents[idx]->treatment) {
-            case SpotValue: cuda11_AllEvents[idx]->cumulativeValue = 0; break;
-            case RunningSum: cuda11_AllEvents[idx]->cumulativeValue = 0; break;
-            case RunningMin: cuda11_AllEvents[idx]->cumulativeValue = DBL_MAX; break;
-            case RunningMax: cuda11_AllEvents[idx]->cumulativeValue = -DBL_MAX; break;
-        }
-
-        // Get the device we need to count it towards.                
-        int eventDeviceNum = c
```

Comment:
I feel like logging every single action is not very useful and only adds unnecessary code

Reply:
?

Reply:
I think I might have commented a stale commit. Disregard this